switch/ogre3d/ogre-switch.patch

```diff
@@ -0,0 +1,690 @@
+--- a/OgreMain/include/OgreString.h
++++ b/OgreMain/include/OgreString.h
+@@ -59,6 +59,16 @@ THE SOFTWARE.
+ #   include <xlocale.h>
+ #endif
+ 
++#if OGRE_PLATFORM == OGRE_PLATFORM_SWITCH
++#   include <stdlib.h>
++#   include <xlocale.h>
++#   include <strings.h>
++
++#   define strtod_l(ptr, end, l) strtod(ptr, end)
++#   define strtoul_l(ptr, end, base, l) strtoul(ptr, end, base)
++#   define strtol_l(ptr, end, base, l) strtol(ptr, end, base)
++#endif
++
+ namespace Ogre {
+     /** \addtogroup Core
+      *  @{
+diff --git a/OgreMain/src/OgreDeflate.cpp b/OgreMain/src/OgreDeflate.cpp
+index 56af65a..cedbdaa 100644
+--- a/OgreMain/src/OgreDeflate.cpp
++++ b/OgreMain/src/OgreDeflate.cpp
+@@ -155,6 +155,8 @@ namespace Ogre
+                 }
+ #elif OGRE_PLATFORM == OGRE_PLATFORM_APPLE_IOS || OGRE_PLATFORM == OGRE_PLATFORM_APPLE
+                 mTempFileName = macTempFileName();
++#elif OGRE_PLATFORM == OGRE_PLATFORM_SWITCH
++                OGRE_EXCEPT(Exception::ERR_INTERNAL_ERROR, "Temporary file name generation failed.", "DeflateStream::init");
+ #else
+                 char tmpname[] = "/tmp/ogreXXXXXX";
+                 if (mkstemp(tmpname) == -1)
+diff --git a/OgreMain/src/OgreFileSystem.cpp b/OgreMain/src/OgreFileSystem.cpp
+index 53b2af5..08446f4 100644
+--- a/OgreMain/src/OgreFileSystem.cpp
++++ b/OgreMain/src/OgreFileSystem.cpp
+@@ -32,7 +32,8 @@ THE SOFTWARE.
+ #if OGRE_PLATFORM == OGRE_PLATFORM_LINUX || OGRE_PLATFORM == OGRE_PLATFORM_APPLE || \
+     OGRE_PLATFORM == OGRE_PLATFORM_APPLE_IOS || \
+     OGRE_PLATFORM == OGRE_PLATFORM_ANDROID || \
+-    OGRE_PLATFORM == OGRE_PLATFORM_EMSCRIPTEN
++    OGRE_PLATFORM == OGRE_PLATFORM_EMSCRIPTEN || \
++    OGRE_PLATFORM == OGRE_PLATFORM_SWITCH
+ #   include "OgreSearchOps.h"
+ #   include <sys/param.h>
+ #endif
+diff --git a/OgreMain/src/OgreRoot.cpp b/OgreMain/src/OgreRoot.cpp
+index d5be1f5..37a5b3e 100644
+--- a/OgreMain/src/OgreRoot.cpp
++++ b/OgreMain/src/OgreRoot.cpp
+@@ -79,6 +79,10 @@ THE SOFTWARE.
+ #include "Android/OgreAndroidLogListener.h"
+ #endif
+ 
++#if OGRE_PLATFORM == OGRE_PLATFORM_SWITCH
++#include <switch.h>
++#endif
++
+ namespace Ogre {
+     //-----------------------------------------------------------------------
+     template<> Root* Singleton<Root>::msSingleton = 0;
+@@ -862,7 +866,11 @@ namespace Ogre {
+         // or break out by calling queueEndRendering()
+         mQueuedEnd = false;
+ 
++#ifdef __SWITCH__
++        while( !mQueuedEnd && appletMainLoop() )
++#else
+         while( !mQueuedEnd )
++#endif
+         {
+             if (!renderOneFrame())
+                 break;
+diff --git a/OgreMain/src/OgreSearchOps.cpp b/OgreMain/src/OgreSearchOps.cpp
+index 48fdc33..1a8dcbd 100644
+--- a/OgreMain/src/OgreSearchOps.cpp
++++ b/OgreMain/src/OgreSearchOps.cpp
+@@ -35,6 +35,12 @@ THE SOFTWARE.
+ #include <string.h>
+ #include <stdlib.h>
+ 
++#if OGRE_PLATFORM == OGRE_PLATFORM_SWITCH
++#if !defined(strdup)
++extern "C" { char* strdup(const char*); }
```

Comment:
Why was it necessary to redefine strdup? We already have strdup...

Reply:
you can use -std=gnu++11 or -D_GNU_SOURCE to enable strdup

Reply:
Hmmm yeah the man page does actually say it's posix and not standard C. They probably shouldn't have used it or told CMake to use the GNU extensions (which they turned OFF at the top of the CMakeLists.txt). Oh well, time to wait for the whole library to slowly build again x.x